docs/detections/detection-engine-intro.asciidoc

```diff
@@ -86,7 +86,7 @@ Indicator match rules provide a powerful capability to search your security data
 
 In addition, the following support restrictions are in place:
 
-* {elastic-sec} does not support the use of either cold or frozen {ref}/data-tiers.html[tier data] with indicator match rules.
+* Indicator match rules don't support cold or frozen data but will query cold and frozen {ref}/data-tiers.html[data tiers] if they match the rule's index pattern. To exclude query results from cold and frozen tiers, create a <<exclude-cold-frozen-data-individual-rules,Query DSL filter>> and add it to individual rules. 
```

Comment:
This part still concerns me because it feels like we're telling users that cold and frozen data aren't supported but the rules by default will put them in this unsupported situation. I think the key points I'd like to communicate are:
- Indicator match rules don't support cold and frozen data
- Normally (when shards are available and timestamps in the data are correct), cold and frozen data should be excluded efficiently by timestamp (i.e. as long as the cold and frozen data is older than the rule interval + additional lookback, it should be fine)
- In abnormal circumstances (shards unavailable and/or bad timestamps) the _tier filter can be added to explicitly exclude cold/frozen data and resolve issues, although for 8.16.0 the _tier filter still isn't sufficient to fix IM rules with unavailable frozen shards so I wouldn't put too much detail about this

Maybe we can write something like "Indicator match rules don't support cold or frozen data. Since rules query the last X=(rule interval + lookback) minutes, indices queried by the rule should wait at least X minutes before moving to cold or frozen tier. Alternatively, to explicitly exclude query results from cold and frozen...".

Reply:
Chatting with Marshall. What do you think if we go with something like:

```suggestion
* Indicator match rules don't support cold or frozen data. Cold or frozen data in indices queried by Indicator Match rules must be older than the time range queried by the rule. If your data's timestamps are unreliable, you can exclude cold and frozen tier data using a <<exclude-cold-frozen-data-individual-rules,Query DSL filter>>. 
```

Please feel free to modify the language to be more grammatically correct :)